.github/workflows/ci.yml

```
@@ -0,0 +1,38 @@
+name: CI Pipeline
+
+on:
+    push:
+    pull_request:
+
+jobs:
+    ci-checks:
+        name: Run CI checks
+        runs-on: ubuntu-latest
+
+        steps:
+            - name: Checkout Repository
+              uses: actions/checkout@v4
+
+            - name: Install Rust
+              uses: dtolnay/rust-toolchain@stable
+              with:
+                components: clippy, rustfmt
+            
+            - name: Format Check
+              run: cargo fmt --check
```

It is better practice to always include the `--all` flag.
```suggestion
              run: cargo fmt --all --check
```